src/backend/gl/src/window/surfman.rs

```diff
@@ -58,12 +57,15 @@ impl Instance {
         raw_handle: raw_window_handle::RawWindowHandle,
     ) -> Surface {
         // Get context attributes
-        let context_attributes = Self::get_default_context_attributes();
+        let mut context_attributes = Self::get_default_context_attributes();
+
+        // Share the context with the root context
+        let root_ctx = &*self.root_context.read();
+        context_attributes.share_with = Some(root_ctx);
 
         // Open a device for the surface
-        // TODO: Assume hardware adapter
         let mut device = SM_CONN
```

Comment:
why are we creating the device here? I thought we'd be using the one we have on the instance already

Reply:
Ah, we probably don't need to. I'll change that.